src/ngraph/op/concat.cpp

```diff
@@ -50,8 +50,10 @@ void op::Concat::validate_and_infer_types()
         Dimension this_input_rank = this_input_shape.rank();
         if (this_input_rank.is_static())
         {
+            m_axis = m_axis < 0 ? m_axis + int64_t(this_input_rank) : m_axis;
```

Comment:
It seems this requires  a static rank and won't work if rank is dynamic

Reply:
The following code is also run for a dynamic_shape if the rank will not be updated then the axis will be equal -1 for each iteration